lib/block/block.go

```diff
@@ -39,6 +39,10 @@ func (bck Block) String() string {
 	return string(encoded)
 }
 
+func (bck Block) IsEmpty() bool {
+	return len(bck.Hash) < 1
```

Comment:
Wait, why do we use the length of the `Hash` ? I would expect it to be always present.

Reply:
I want to check the empty `block.Block` from `block.Block{}`. We can add another check points.

Reply:
An empty `Block` is invalid, isn't it ? In this case I'd rename this method to `IsInvalid`.